server/aws-lsp-codewhisperer/src/language-server/localProjectContext/initializeUtils.ts

```diff
@@ -0,0 +1,23 @@
+import { InitializeParams, WorkspaceFolder } from '@aws/language-server-runtimes/protocol'
+import * as path from 'path'
+import { URI } from 'vscode-uri'
+
+export function getWorkspaceFolders(params?: InitializeParams): WorkspaceFolder[] {
+    if (!params) {
+        return []
+    }
+    const getFolderName = (uri: string) => path.basename(URI.parse(uri).path) || 'workspace'
+    if (params.workspaceFolders) {
```

Comment:
Also check if the array is empty, so that we avoid returning an empty list too early